src/adaptor/native.ts

```diff
@@ -0,0 +1,30 @@
+import nativeFirestore, {
+  FirebaseFirestoreTypes
+} from '@react-native-firebase/firestore'
+// @ts-ignore: React Native Firebase doesn't export types for internal modules
+import FirestoreDocumentReference from '@react-native-firebase/firestore/lib/FirestoreDocumentReference'
```

Comment:
I think you could use Firestore types without @ts-ignore

```
import { FirebaseFirestoreTypes } from "@react-native-firebase/firestore";

type DocRef = FirebaseFirestoreTypes.DocumentReference
type FieldRef = FirebaseFirestoreTypes.FieldValue
```

All RN Firestore types are grouped into FirebaseFirestoreTypes namespace.

Reply:
These aren't types but constructors that I use internally for example `adaptor.consts.FieldValue.delete()` or `data instanceof adaptor.consts.DocumentReference` so sadly it won't fly.

Reply:
so for actual values, you'll need

```
import firestore, { FirebaseFirestoreTypes, firebase } from '@react-native-firebase/firestore';
firebase.firestore.FieldValue.delete()
```

etc